state/execution.go

```diff
@@ -123,14 +129,47 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 		maxReapBytes = -1
 	}
 
-	txs := blockExec.mempool.ReapMaxBytesMaxGas(maxReapBytes, maxGas)
+	// check if oracle's gossipVoteMap has any results
+	blockExec.oracleInfo.GossipVoteBuffer.UpdateMtx.RLock()
+	oracleVotesBuffer := blockExec.oracleInfo.GossipVoteBuffer.Buffer
+	blockExec.oracleInfo.GossipVoteBuffer.UpdateMtx.RUnlock()
+
+	var signGossipVoteTxBz []byte
+	if len(oracleVotesBuffer) > 0 {
+		votes := []*oracleproto.GossipVote{}
+		for _, vote := range oracleVotesBuffer {
+			votes = append(votes, vote)
+		}
+
+		resp, err := blockExec.proxyApp.SignGossipVote(ctx, &abci.RequestSignGossipVote{
+			ProposerAddress: proposerAddr,
+			GossipVotes:     votes,
```

Comment:
instead of signing gossipvotes, sign a more compressed version of votes. can bitmap []validators. and reconstruct gossip votes.